### PR TITLE
Use fs:proximate() due to missing fs::relative()

### LINKED ISSUE #2703: Issues with building opm-common
I just pulled master and tried to build opm-common but it fails. 

```
/home/thsa/workspace/opm/opm-common/src/opm/parser/eclipse/Parser/Parser.cpp: In member function ‘Opm::Deck Opm::Parser::parseFile(const string&, const Opm::ParseContext&, Opm::ErrorGuard&, const std::vector<Opm::Ecl::SectionType>&) const’:
/home/thsa/workspace/opm/opm-common/src/opm/parser/eclipse/Parser/Parser.cpp:1187:42: error: ‘relative’ is not a member of ‘Opm::filesystem’
             data_file = Opm::filesystem::relative( Opm::filesystem::canonical(dataFileName) );
```
Any clue? Jenkins seems to work.

### PR DESCRIPTION
**Pull request by joakim-hove**
Resolves #2703

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/parser/eclipse/Parser/Parser.cpp [hunk 1]
```diff
@@ -1184,7 +1184,7 @@ bool parseState( ParserState& parserState, const Parser& parser ) {
         if (dataFileName[0] == '/')
             data_file = Opm::filesystem::canonical(dataFileName).string();
         else
-            data_file = Opm::filesystem::relative( Opm::filesystem::canonical(dataFileName) );
+            data_file = Opm::filesystem::proximate( Opm::filesystem::canonical(dataFileName) );
```
Comment (totto82) [inline src/opm/parser/eclipse/Parser/Parser.cpp:1188]: I think you need to remove filesystem. Opm::proximate

Comment (joakim-hove) [reply src/opm/parser/eclipse/Parser/Parser.cpp:1188]: OK - have you tried? The current PR works on my machine - for what it is worth

Comment (akva2) [reply src/opm/parser/eclipse/Parser/Parser.cpp:1188]: you have to use Opm::proximate. You are using the one exposed in the filesystem namespace which is not available with the ts.

Comment (totto82) [reply src/opm/parser/eclipse/Parser/Parser.cpp:1188]: If I use Opm::proximate everything build fine

Comment (joakim-hove) [reply src/opm/parser/eclipse/Parser/Parser.cpp:1188]: OK - updated

**Comment by joakim-hove** [comment]
jenkins build this please

**Pull request merged**